internal/controller/ols_app_redis_assets.go

```diff
@@ -70,28 +126,71 @@ func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig)
 	return &deployment, nil
 }
 
-func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
-	DeploymentSelectorLabels := map[string]string{
-		"app.kubernetes.io/component":  "redis-server",
-		"app.kubernetes.io/managed-by": "lightspeed-operator",
-		"app.kubernetes.io/name":       "lightspeed-service-redis",
-		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+// updateRedisDeployment updates the deployment based on CustomResource configuration.
+func (r *OLSConfigReconciler) updateRedisDeployment(ctx context.Context, existingDeployment, desiredDeployment *appsv1.Deployment) error {
+	changed := false
+
+	// Validate deployment annotations.
+	if existingDeployment.Annotations == nil ||
+		existingDeployment.Annotations[OLSRedisConfigHashKey] != r.stateCache[OLSRedisConfigHashStateCacheKey] {
+		updateDeploymentAnnotations(existingDeployment, map[string]string{
+			OLSRedisConfigHashKey: r.stateCache[OLSRedisConfigHashStateCacheKey],
+		})
+		// update the deployment template annotation triggers the rolling update
+		updateDeploymentTemplateAnnotations(existingDeployment, map[string]string{
+			OLSRedisConfigHashKey: r.stateCache[OLSRedisConfigHashStateCacheKey],
+		})
+
+		changed = true
 	}
+
+	// Validate volume mounts for a specific container in deployment.
```

Comment:
Appears not right comment here  @vishnuchalla

Reply:
Updated the comment.